tests/datatypes/test_list.py

```diff
@@ -119,6 +119,62 @@ def test_append(self):
             print(x[4], x[5])
             """)
 
+    def test_extend(self):
+        # extend a list
+        self.assertCodeExecution("""
+            x = [1, 2, 3]
+            x.extend([4, "hello"])
+            print(x)
+            """)
+
+        # extend a tuple
+        self.assertCodeExecution("""
+            x = [1, 2, 3]
+            x.extend((5, "world", "hello"))
+            print(x)
+            """)
+
+        # extend a frozenset
+        self.assertCodeExecution("""
+            x = [1, 2, 3]
+            y = frozenset([8, "theta"])
+            x.extend(y)
+            set([x[3], x[4]]) == y
```

Comment:
I can see what you're trying for here, but there are two problems:

Firstly, this won't actually test anything. `assertCodeExecution()` performs it's test based on the output of the program. It's not enough to just do the equality test - you actually have to print something, so that there's something in the output to compare.

Secondly, if the problem you're getting is output ordering - we've got a mechanism in place to work around this. Any TranspileTestCase instance can define a "substitutions" dictionary. If present, any "values" in the dictionary that are present in the sample output will be replaced by it's "key". Essentially, this normalises unpredictable output so that it will reliably pass tests.

There's an example in the InplaceListOperationTests class in the same file that programatically generates a substitutions dictionary, but for simpler cases, you can statically define them, too.

Reply:
Oops, yes forgot the `print`.

Ah, I see how you are doing it, will take a crack at it tomorrow.

Thanks for the feedback.

Reply:
Added a new function here to create the substitutions for those tests